spec/stringProcessing/linkTypeSpec.js

```diff
@@ -1,14 +1,52 @@
 var linkTypeFunction = require( "../../js/stringProcessing/getLinkType.js" );
 
-describe( "checks type of link", function(){
+describe( "getLinkType", function(){
+
 	it( "should classify relative links as internal", function() {
-		expect( linkTypeFunction( "<a href='http://example.org/'>Internal</a>", "http://example.org" ) ).toBe( "internal" );
-		expect( linkTypeFunction( "<a href='http://example.org/some-path?query_args#hash'>Internal</a>", "http://example.org" ) ).toBe( "internal" );
+		expect( linkTypeFunction( "<a href='another-path'>Internal</a>", "http://example.org" ) ).toBe( "internal" );
+	} );
+
+	it( "should classify relative links with query params as internal", function() {
+		expect( linkTypeFunction( "<a href='another-path?test=123&another=abc'>Internal</a>", "http://example.org" ) ).toBe( "internal" );
+	} );
+
+	it( "should classify relative links with empty query params as internal", function() {
+		expect( linkTypeFunction( "<a href='another-path?test=&another='>Internal</a>", "http://example.org" ) ).toBe( "internal" );
+	} );
+
+	it( "should classify relative links with fragments as internal", function() {
+		expect( linkTypeFunction( "<a href='another-path#subheader'>Internal</a>", "http://example.org" ) ).toBe( "internal" );
+	} );
+
+	it( "should classify absolute links without host as internal", function() {
```

Comment:
I'm missing tests for `href='some-path'`, `href="/path/to"` and `href="../path/to"`

Reply:
Added